plugin/makeshift.vim

```diff
@@ -87,6 +88,12 @@ function! s:makeshift()
     if len(l:program) == 0
         let l:program = s:determine_build_system(expand('%:p:h'))
     endif
+    if l:program == 'waf'
+        let l:found = globpath(b:makeshift_root, l:program)
+        if filereadable(l:found)
+            let l:program = l:found
+        endif
+    endif
```

Comment:
Thanks for your feedback! A bit of background; I'm working on multiple repositories/libraries with different build systems. Most waf based libraries bundle waf, but I also have waf installed on my system and in some cases waf isn't bundled.

Anyways, I found makeshift to increase my productivity switching between repositories, rather than modifying makeprg myself. And I just quickly hacked this in to make it work for me. I do like your approach better, and will test it today. Thanks again!